apps/test-app/frontend/src/components/tree-widget/StatelessTree.tsx

```diff
@@ -89,22 +89,17 @@ function Tree({ imodel, imodelAccess, height, width }: { imodel: IModelConnectio
     setFormatter(newValue ? customFormatter : undefined);
   };
 
-  const { getHierarchyLevelDetails } = treeProps;
-  const [filteringOptions, setFilteringOptions] = useState<{ nodeId: string | undefined; hierarchyLevelDetails: HierarchyLevelDetails }>();
-  const onFilterClick = useCallback(
-    (nodeId: string | undefined) => {
-      const hierarchyLevelDetails = getHierarchyLevelDetails(nodeId);
-      setFilteringOptions(hierarchyLevelDetails ? { nodeId, hierarchyLevelDetails } : undefined);
-    },
-    [getHierarchyLevelDetails],
-  );
+  const [filteringOptions, setFilteringOptions] = useState<HierarchyLevelDetails>();
+  const onFilterClick = useCallback((hierarchyLevelDetails: HierarchyLevelDetails) => {
+    setFilteringOptions(hierarchyLevelDetails);
+  }, []);
```

Comment:
I think this is not needed any more. You can pass `setFilteringOptions` directly.

Reply:
Right, the `nodeId` is no longer stored. Fixed.